### ISSUE TITLE
`anonymous-formation` incorrectly flags valid multi-segment dot notation as undefined object access

### ISSUE THREAD

**Issue by ImilB** [objectionary/lints#1154]
## Problem

The `anonymous-formation` lint treats valid access chains inside anonymous formations as undefined when the chain contains more than one dot segment after a local void attribute.

The current matching logic only allows a single optional suffix after the void attribute name, so a chain like `i.put.plus` is not handled correctly.

## What happens

A valid access chain starting from a local void attribute inside an anonymous formation is reported as if it were accessing an undefined object.

This should be accepted, because i is a local void attribute of the anonymous formation.
Why this is a bug
The rule is intended to reject access to undefined outer objects from inside anonymous formations.
It should not reject valid chained access that starts from a defined local attribute. The current implementation conflates these two cases because the pattern only supports one optional suffix segment.

For example, in this EO snippet:

```eo
x > @
  [i]
    seq > @
      *
        i.put.plus 1

**Comment by h1alexbel** [comment]
@ImilB this case passes with no defects:

```eo
# Main.
[] > main
  x > @
    [i]
      seq > @
        *
          i.put.plus 1
```

Am I missing something?

**Comment by 0crat** [comment]
@imilb Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+6](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
